### ISSUE TITLE
aria-disabled is not added to disabled button

### ISSUE THREAD

**Issue by unional** [adobe/react-spectrum#8825]
### Provide a general summary of the issue here

```tsx
<Button isDisabled onPress={() => alert("Hello world!")}>
  Press me
</Button>
```

should add `aria-disabled` in html. It does not.

`aria-disabled` is mentioned in: https://react-spectrum.adobe.com/react-aria/Button.html#props

### 🤔 Expected Behavior?

should add `aria-disabled`.

And we cannot add that manually because `react-aria-components` control all `aria-*` and `data-*` attributes.

### 😯 Current Behavior

It does not.

### 💁 Possible Solution

Fix the bug

### 🖥️ Steps to Reproduce

https://codesandbox.io/p/sandbox/summer-morning-7wqs77

### Version

1.11.0

### What browsers are you seeing the problem on?

Chrome

### What operating system are you using?

mac os

### 🧢 Your Company/Team

Palo Alto Networks

**Comment by LFDanLu** [comment]
The important distinction to make here between `disabled` and `aria-disabled` is that `disabled` will make the control non-interactive (aka can't be clicked, tabbed to, etc) and thus is sufficient for a typical use case. However, if you want to keep the button discoverable by screen readers when tabbing through the page (note that VoiceOver will allow you to navigate to the `isDisabled` via Control + Option + Arrow Keys), you can make use of `aria-disabled` and apply additional logic to ensure the button doesn't fire on press. This is discussed in brief in https://www.deque.com/blog/distinguishing-between-aria-and-native-html-attributes. 

As an aside, VoiceOver also announces a `disabled` button and `aria-disabled` button exactly the same (i.e. as a dimmed button) so the information conveyed is same and doesn't require both to be applied.
